### PR TITLE
Additional tests

### PR DESCRIPTION
**Pull request by semuserable**
WIP

* Debouncer tests added based on real time
* `UnityApplicationLoggingIntegration` more tests based on types

#skip-changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## Unreleased
+
+- Additional tests ([#232](https://github.com/getsentry/sentry-unity/pull/232))
```
Comment (bruno-garcia) [inline CHANGELOG.md:5]: This doesn't really impact the customer so feel free to use `#skip-changelog` in such cases

File: test/Sentry.Unity.Tests/DebouncerTests.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+using UnityEngine.TestTools;
+
+namespace Sentry.Unity.Tests
+{
+    /// <summary>
+    /// Testing real in realtime.
+    /// </summary>
+    public sealed class DebouncerTests
+    {
+        private readonly TimeSpan DefaultOffset = TimeSpan.FromSeconds(1);
```
Comment (bruno-garcia) [inline test/Sentry.Unity.Tests/DebouncerTests.cs:14]: Ideally we mock any 'time based' test as they tend to:

1. Slow down builds
2. Be flaky

If there's no way around it, can we rely on timing such as:

```suggestion
        private readonly TimeSpan DefaultOffset = TimeSpan.FromMilliseconds(1);
```

File: test/Sentry.Unity.Tests/UnityApplicationLoggingIntegrationTests.cs [hunk 3]
```diff
@@ -56,5 +57,42 @@ public void OnLogMessageReceived_Breadcrumbs_Added()
 
             Assert.AreEqual(2, hub.ConfigureScopeCalls.Count);
         }
+
+        [TestCaseSource(nameof(LogTypesAndSentryLevels))]
+        public void OnLogMessageReceived_UnityErrorLogTypes_CapturedAndCorrespondToSentryLevel(LogType unityLogType, SentryLevel sentryLevel)
+        {
+            var sut = _fixture.GetSut();
+            var hub = _fixture.Hub;
+            sut.Register(hub, SentryOptions);
```
Comment (bruno-garcia) [inline test/Sentry.Unity.Tests/UnityApplicationLoggingIntegrationTests.cs:66]: Couldn't this:
```
            var hub = _fixture.Hub;
            sut.Register(hub, SentryOptions);
```
be part of the fixture? In `GetSut`?

```suggestion
            var sut = _fixture.GetSut();
```

**Comment by semuserable** [comment]
#skip-changelog

**Comment by semuserable** [comment]
Unity license failed

File: test/Sentry.Unity.Tests/UnityApplicationLoggingIntegrationTests.cs (hunk 3 above)
Comment (semuserable) [reply test/Sentry.Unity.Tests/UnityApplicationLoggingIntegrationTests.cs:66]: Are you looking at something like this

```csharp
private class Fixture
{
    public TestHub Hub  { get; }

    public UnityApplicationLoggingIntegration Sut { get; }

    public Fixture()
    {
        Hub = new TestHub();
        Sut = new UnityApplicationLoggingIntegration(new TestApplication());
    }
}

[Test]
void Test()
{
  var fixture = new Fixture();
  fixture.Sut.Register(fixture.Hub, SentryOptions);

  fixture.Sut.OnLogMessageReceived("condition", "stacktrace", unityLogType);

  Assert.AreEqual(0, fixture.Hub.CapturedEvents.Count);
}
```

File: test/Sentry.Unity.Tests/DebouncerTests.cs (hunk 2 above)
Comment (semuserable) [reply test/Sentry.Unity.Tests/DebouncerTests.cs]: Made 100ms

File: test/Sentry.Unity.Tests/UnityApplicationLoggingIntegrationTests.cs [hunk 4]
```diff
@@ -10,64 +10,67 @@ public sealed class UnityApplicationLoggingIntegrationTests
...
+            Assert.AreEqual(1, _hub.CapturedEvents.Count);
         }
 
         [Test]
         public void OnLogMessageReceived_WithSeveralErrorsDebounced_CaptureEvent()
         {
-            var sut = _fixture.GetSut();
-            var hub = _fixture.Hub;
-            sut.Register(hub, SentryOptions);
+            var sut = _fixture.GetSut(_hub, _sentryOptions);
```
Comment (bruno-garcia) [inline test/Sentry.Unity.Tests/UnityApplicationLoggingIntegrationTests.cs:47]: Couldn't this be part of the Fixture? This way you don't need to pass it here

File: test/Sentry.Unity.Tests/UnityApplicationLoggingIntegrationTests.cs [hunk 5]
```diff
@@ -10,64 +10,67 @@ public sealed class UnityApplicationLoggingIntegrationTests
...
         private class Fixture
         {
-            public Stubs.TestHub Hub => new();
-            public IApplication Application => new TestApplication();
-
-            public UnityApplicationLoggingIntegration GetSut() => new(Application);
+            public UnityApplicationLoggingIntegration GetSut(IHub hub, SentryOptions sentryOptions)
+            {
+                var application = new TestApplication();
+                var integration = new UnityApplicationLoggingIntegration(application);
```
Comment (bruno-garcia) [inline test/Sentry.Unity.Tests/UnityApplicationLoggingIntegrationTests.cs:16]: These can be properties. so you can mutate them (to set things up) before calling `GetSut`

**Comment by semuserable** [comment]
Updated. Breadcrumb tests added; Fixture refactor.

cc @bruno-garcia

**Review by bruno-garcia — APPROVED** [review]
Awesome.
Thanks Dmytro!

**Pull request merged**